### PR TITLE
Add mysql conformance test

### PR DESCRIPTION
**Pull request by beiwei30**
# Description

Add conformance test for mysql state, and make mysql pass the test.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #1122 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: state/mysql/mysql_test.go [hunk 1]
```diff
@@ -308,61 +308,6 @@ func TestDeleteWithETag(t *testing.T) {
 	assert.Nil(t, err)
 }
 
-func TestReturnNDBResultsRowsAffectedReturnsError(t *testing.T) {
```
Comment (CodeMonkeyLeet) [inline state/mysql/mysql_test.go:311]: Is there a way for us to preserve the unit tests? The mock tests can be useful for dev inner loop testing and with the exception of the TooManyRows check, the rest seem like they could be updated to check the updated behavior.

File: .github/infrastructure/docker-compose-mysql.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    image: mysql
+    command: --default-authentication-plugin=mysql_native_password
+    restart: always
+    environment:
+      MYSQL_ROOT_PASSWORD: root
+      MYSQL_DATABASE: dapr_state_store
+      MYSQL_USER: dapr
+      MYSQL_PASSWORD: example
+    ports:
+      - "3306:3306"
```
Comment (CodeMonkeyLeet) [inline .github/infrastructure/docker-compose-mysql.yml:13]: Nit: newline at end of Linux text file

File: state/mysql/mysql.go [hunk 3]
```diff
@@ -242,7 +242,7 @@ func (m *MySQL) ensureStateTable(stateTableName string) error {
 		// in on inserts and updates and is used for Optimistic Concurrency
 		createTable := fmt.Sprintf(`CREATE TABLE %s (
 			id varchar(255) NOT NULL PRIMARY KEY,
-			value json NOT NULL,
+			value text NOT NULL,
```
Comment (CodeMonkeyLeet) [inline state/mysql/mysql.go:245]: For my understanding, what breaks when this column is set to `json`?

**Review by CodeMonkeyLeet — COMMENTED** [review]
Thanks so much for adding the conformance test for mysql state store, these tests are much needed!

File: state/mysql/mysql.go (hunk 3 above)
Comment (beiwei30) [reply state/mysql/mysql.go:245]: pls. check https://github.com/dapr/components-contrib/pull/1118#discussion_r704013615

File: .github/infrastructure/docker-compose-mysql.yml (hunk 2 above)
Comment (beiwei30) [reply .github/infrastructure/docker-compose-mysql.yml]: fixed.

File: state/mysql/mysql_test.go (hunk 1 above)
Comment (beiwei30) [reply state/mysql/mysql_test.go:311]: I add more unit tests to serve the same purpose.

**Review by CodeMonkeyLeet — APPROVED** [review]
LGTM, love the addition of more tests and fixing up issues learned from the postrgres state component!

**Comment by yaron2** [comment]
Merged. Thanks for this @beiwei30

**Pull request merged**